cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/DiagnosticsProcessor.java

```diff
@@ -102,14 +117,30 @@ private Module createModule(final ModuleConfiguration moduleConfiguration) throw
     }
 
     /**
-     * Process a query with all configured modules.
+     * Accepts a query sent by the connector. with all configured modules.
      *
      * @param query query to process
      */
     public void process(final Query query) {
-        logger.trace("Processing query {}", query);
-        for (Module module : modules) {
-            module.process(query);
+        logger.trace("Accepting query {}", query);
+        if (!queryBufferQueue.offer(query)) {
+            // the queue is full
+            logger.warn("The query buffer queueu is full. The reported query is ignored.");
```

Comment:
Logging only once could pass unnoticed. A combination of a flag and a log threshold hysteresis can be added to avoid excessive logging. Would it be ok?